### PR TITLE
github - user api

### PR DESCRIPTION
**Pull request by vinaynv3**
**Implemented GraphQL server using Django graphQL library called graphene (https://graphene-python.org/).**

1.  Subclassed graphene module to create query root for GraphQL server
2.  Implemented GithubAPI module to handle requests from external API (React)
3.  classes in the above GithubAPI module will make the API(restAPI) calls to Github API using the requests
     (https://docs.python-requests.org/en/master/) module for responses.

### CODE REVIEW

File: backend/DataEntry/githubAPI.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+
+"""
+Below implentation requests github user's profile and and repository information
+using API https://api.github.com (version v3)
+"""
+
+from datetime import date
+import requests # HTTPS library
```
Comment (saranshvaid8) [inline backend/DataEntry/githubAPI.py:8]: comments on imports don't look good

File: backend/DataEntry/githubAPI.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    # make api call
+    @functools.lru_cache() # last recently used cache decorator
+    def get_user_object(self):
+
+        github_api = 'https://api.github.com'
+        url = '{0}/users/{1}'.format(github_api,self.username)
+        data = requests.get(url)
+        return data
+
+    #date formater
```
Comment (saranshvaid8) [inline backend/DataEntry/githubAPI.py:28]: casing is not consistent

File: backend/DataEntry/githubAPI.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                    'joined' : self.format_date(self.data.setdefault('created_at',None)),
+                    'details' : self.usr_details()
+                    }
+            return profile
+
+        else:
+            return Exception("username {0} not found in github, \
+                            please verify!".format(self.username))
+
+    # return user data to GraphQL schema
```
Comment (saranshvaid8) [inline backend/DataEntry/githubAPI.py:57]: consistent casing on comments

**Review by saranshvaid8 — CHANGES_REQUESTED** [review]
Please resolve the comments

File: backend/DataEntry/githubAPI.py (hunk 1 above)
Comment (vinaynv3) [reply backend/DataEntry/githubAPI.py:8]: re-designing backend from the core, above code base, is not part of current designs.
resolved the issue as closed.

**Review by saranshvaid8 — APPROVED** [review]

**Pull request merged**